### ISSUE TITLE
Issue with v1.0.4 on Craft 3 - "Your requirements could not be resolved to an installable set of packages."

### ISSUE THREAD

**Issue by iamkeir** [codewithkyle/craft-readonly#5]
Hi there,

We have a Craft 3 install prompting us to upgrade to v1.0.4 which we can do so seemingly successfully.

However, if we ever run `composer update` afterwards, we receive:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires codewithkyle/read-only 1.0.4, found codewithkyle/read-only[dev-master, 1.0.0, 1.0.1, 1.0.2, 1.0.5] but these do not match your constraint and are therefore not installable. Make sure you either fix the constraint or avoid updating this package to keep the one present in the lock file (codewithkyle/read-only[1.0.4]).
```

If we roll our `composer.json` back to 1.0.2, this fixes the composer update issue, but Craft then starts prompting us to upgrade to v1.0.4 again.

So we're trapped in a version loop.

Ref: https://github.com/codewithkyle/craft-readonly/issues/4

**Comment by codewithkyle** [comment]
It appears that the update prompt comes from Craft's API: https://api.craftcms.com/v1/updates

A GET request is made when the `/utilities/updates` page is loaded. The request headers contain the current install version of each plugin in the `x-craft-system` header.

![image](https://user-images.githubusercontent.com/15202776/217646428-eec573ec-e4f3-49d2-bd60-7c57f801b231.png)

Since their API is not open source I can't know for sure but I'm guessing that they cache released version numbers & notes when a new package version is pushed to packagist. The updates page then renders the response.

![image](https://user-images.githubusercontent.com/15202776/217647069-701dbc65-3fb0-40fd-af25-68fdc960af2f.png)

Next steps:
- verify the request headers contain a reference to `read-only:1.0.2`
- verify the response contains an update to the non-existent "1.0.4" package version

**Comment by iamkeir** [comment]
It seems not to be prompting for it anymore - maybe Craft's cache cleared?

**Comment by iamkeir** [comment]
No, I take that back - it's now offering an update for Craft 4 even though we're still on Craft 3.
![screenshot_691](https://user-images.githubusercontent.com/539615/217877803-6b9b237b-103d-48c6-bc0a-e34fb8f07c14.png)

**Comment by codewithkyle** [comment]
> It seems not to be prompting for it anymore - maybe Craft's cache cleared?

That's great to hear. Now I don't have to open a ticket with them.

As for the Read Only v1.0.3...

No new fixes/features were added in 1.0.3 or 1.0.4 in this plugin. I was trying to handle the the Craft 3 PHP 7 -> Craft 3 PHP 8 -> Craft 4 PHP 8 upgrade path, clearly I screwed it up.

I've dropped v1.0.3 from packagist so it should soon be removed from the Craft update API soon.

The last version that should be available for Craft 3 will be v1.0.2 and it should work for both PHP 7 and PHP 8. The 1.0.5+ versions should only be available for Craft 4. This should hopefully be the last change required to close both this issue and #4.
